src/components/global/Playground/index.tsx

```diff
@@ -0,0 +1,430 @@
+        'src/components/Example.vue': codeBlock,
+        'public/index.html': '<div id="app"></div>',
+      },
+      dependencies: {
+        vue: '^3.0.0',
+        'vue-router': '^4.0.12',
+        '@ionic/vue': DEFAULT_IONIC_VERSION,
+        // Stackblitz requires this dependency to run
+        '@stencil/core': '^2.13.0',
+      },
+    });
+  }
+
+  function copySourceCode() {
+    const copyButton = codeRef.current.querySelector('button');
+    copyButton.click();
+  }
+
+  function openEditor(event) {
+    const button = event.target.closest('button');
+    // Outer text includes line breaks `\n` to maintain code formatting in editor examples.
+    const codeBlock = button.parentElement.parentElement.querySelector('.code-block pre code')?.outerText;
+    const editorOptions = {
+      title,
+      description,
+    };
+    switch (selected) {
+      case UsageTarget.Angular:
+        openAngularEditor(codeBlock, editorOptions);
+        break;
+      case UsageTarget.Basic:
+        openHtmlEditor(codeBlock, editorOptions);
+        break;
+      case UsageTarget.React:
+        openReactEditor(codeBlock, editorOptions);
+        break;
+      case UsageTarget.Vue:
+        openVueEditor(codeBlock, editorOptions);
+        break;
+    }
+  }
+
+  const languages = UsageTargetList.filter((lang) => Object.keys(codeSnippet).includes(lang.toLowerCase()));
+
+  return (
+    <div className={`playground playground--mode-${mode}`}>
+      <div className="playground__container">
+        <div className="playground__control-toolbar">
+          <div className="playground__control-group">
+            {languages.map((language) => (
+              <CodeBlockButton key={language} language={language} selected={selected} setSelected={setSelected} />
+            ))}
+          </div>
+          <div className="playground__control-group">
+            <button
+              type="button"
+              className={
+                'playground__control-button ' + (mode === Mode.iOS ? 'playground__control-button--selected' : '')
+              }
+              onClick={() => setMode(Mode.iOS)}
+            >
+              iOS
+            </button>
+            <button
+              type="button"
+              className={
+                'playground__control-button ' + (mode === Mode.MD ? 'playground__control-button--selected' : '')
+              }
+              onClick={() => setMode(Mode.MD)}
+            >
+              MD
+            </button>
+          </div>
+          <div className="playground__control-group playground__control-group--end">
+            {/* Toggle Source Code */}
+            <Tippy content={expanded ? 'Hide source code' : 'Show full source'}>
+              <button
+                className="playground__icon-button"
+                aria-label={expanded ? 'Hide source code' : 'Show full source'}
+                onClick={() => setExpanded(!expanded)}
+              >
+                <svg
```

Comment:
This icon is giving me troubles exporting cleaning from Figma. We will need to eventually swap this out with a revised SVG viewbox & path that matches the design.